lib/class-wp-theme-json-gutenberg.php

```diff
@@ -289,6 +310,28 @@ public function __construct( $theme_json = array() ) {
 		$valid_block_names   = array_keys( self::get_blocks_metadata() );
 		$valid_element_names = array_keys( self::ELEMENTS );
 		$this->theme_json    = self::sanitize( $theme_json, $valid_block_names, $valid_element_names );
+
+		// Internally, presets are keyed by origin.
+		$nodes = self::get_setting_nodes( $this->theme_json );
+		foreach ( $nodes as $node ) {
+			foreach ( self::PRESETS_METADATA as $preset ) {
```

Comment:
Not all presets are keyed by origin, only colors, gradients, font sizes, and font families are. For example, duotone is not. I think we need to take that into account inside the look.

Reply:
`duotone` is not part of `PRESETS_METADATA` as far as I can see?

Reply:
should all presets be keyed? Why have special cases?

Reply:
> `duotone` is not part of `PRESETS_METADATA` as far as I can see?

Yes, I wrongly thought it was part of presets metadata.

> should all presets be keyed? Why have special cases?

The rationale was that duotone was dynamically injected there was no need to output class variables etc.  Also, duotone does not references or contains a name the direct colors are store in attributes `"duotone":["#000097","#ff4747]`. We don't have a need for this system there. But for consistency reasons, we may reconsider.